client/mana.go

```diff
@@ -0,0 +1,126 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+
+	webapi_mana "github.com/iotaledger/goshimmer/plugins/webapi/mana"
+)
+
+const (
+	routeGetMana                  = "mana"
+	routeGetAllMana               = "mana/all"
+	routeGetManaPercentile        = "mana/percentile"
+	routeGetOnlineAccessMana      = "mana/access/online"
+	routeGetOnlineConsensusMana   = "mana/consensus/online"
+	routeGetNHighestAccessMana    = "mana/access/nhighest"
+	routeGetNHighestConsensusMana = "mana/consensus/nhighest"
+)
+
+// GetOwnMana returns the access and consensus mana of the node this api client is communicating with.
+func (api *GoShimmerAPI) GetOwnMana() (*webapi_mana.GetManaResponse, error) {
+	res := &webapi_mana.GetManaResponse{}
+	if err := api.do(http.MethodGet, routeGetMana,
+		&webapi_mana.GetManaRequest{Node: ""}, res); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// GetManaFullNodeID returns the access and consensus mana of the node specified in the argument.
+// Note, that for the node to understand which nodeID we are referring to, short node ID is not sufficient.
+func (api *GoShimmerAPI) GetManaFullNodeID(fullNodeID string) (*webapi_mana.GetManaResponse, error) {
+	res := &webapi_mana.GetManaResponse{}
+	if err := api.do(http.MethodGet, routeGetMana,
+		&webapi_mana.GetManaRequest{Node: fullNodeID}, res); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// GetMana returns the access and consensus mana a node hase based on its shortNodeID.
+func (api *GoShimmerAPI) GetMana(shortNodeID string) (*webapi_mana.GetManaResponse, error) {
+	// ask the node about the full mana map and filter out based on shortID
+	allManaRes := &webapi_mana.GetAllManaResponse{}
+	if err := api.do(http.MethodGet, routeGetAllMana,
+		nil, allManaRes); err != nil {
+		return nil, err
+	}
+	res := &webapi_mana.GetManaResponse{Node: shortNodeID}
+	// look for node's mana values in the map
+	for _, nodeStr := range allManaRes.Access {
+		if nodeStr.ID == shortNodeID {
+			res.Access = nodeStr.Mana
+			break
+		}
+	}
+	for _, nodeStr := range allManaRes.Consensus {
+		if nodeStr.ID == shortNodeID {
+			res.Consensus = nodeStr.Mana
+			break
+		}
+	}
+	return res, nil
```

Comment:
> If the access and consensus mana lists always have the same length, then you can use 1 index for loop over the range of either of them. - I think their lengths are the same.

their length can be different, it's just the same now because we always use the same pledgeID for both access and consensus.

> Don't we need GetAllMana or at least a version of it to return full ID strings? Coz its a bit weird to get results in short IDs but query the API with long ID only.

Yeah, maybe its for the better if we always include `nodeID` and `fullNodeID` in the response.